src/windows-emulator/syscalls.cpp

```diff
@@ -3413,6 +3413,84 @@ namespace
         c.win_emu.yield_thread();
         return STATUS_SUCCESS;
     }
+    typedef struct _UNICODE_STRING
+    {
+        USHORT Length;
+        USHORT MaximumLength;
+        PWSTR Buffer;
+    } UNICODE_STRING;
+    typedef UNICODE_STRING* PUNICODE_STRING;
+    typedef struct _CLSMENUNAME
+    {
+        LPSTR pszClientAnsiMenuName;
+        LPWSTR pwszClientUnicodeMenuName;
+        PUNICODE_STRING pusMenuName;
+    } CLSMENUNAME, *PCLSMENUNAME;
+
+    NTSTATUS handle_NtUserRegisterClassExWOW(const syscall_context& c, const emulator_object<WNDCLASSEXW> lpwcx,
+                                             emulator_object<UNICODE_STRING> pustrClassName,
+                                             emulator_object<UNICODE_STRING> ClsNVersion,
+                                             emulator_object<CLSMENUNAME> pClassMenuName)
+    {
+        WNDCLASSEXW wndClassEx = lpwcx.read();
+        int index = 0;
+
+        std::wstring name{};
+        std::wstring tempString;
+        int i = 0;
+
+        while (true)
+        {
+            const auto character = c.emu.read_memory<wchar_t>(wndClassEx.lpszClassName + i);
+
+            if (character == L'\0')
+            {
+                break;
+            }
+            tempString += character;
+            i++;
+        }
+
+        name = tempString.c_str();
+
+        if (!c.proc.atoms.empty())
+        {
+            auto i = c.proc.atoms.end();
+            --i;
+            index = i->first + 1;
+        }
+
+        std::optional<uint16_t> last_entry{};
+        for (auto& entry : c.proc.atoms)
+        {
+            if (entry.second == name)
+            {
+                return entry.first;
+            }
+
+            if (entry.first > 0)
+            {
+                if (!last_entry)
+                {
+                    index = 0;
+                }
+                else
+                {
+                    const auto diff = entry.first - *last_entry;
+                    if (diff > 1)
+                    {
+                        index = *last_entry + 1;
+                    }
+                }
+            }
+
+            last_entry = entry.first;
+        }
+        c.win_emu.log.print(color::gray, "Full ClassName: %ls , atom : %u\n", name.c_str(), index);
```

Comment:
feels like a lot of code is duplicated from NtAddAtomEx. can this be unified? e.g. this calls handle_NtAddAtom or can the logic be extracted?